Kaem/kaem.c

```diff
@@ -553,8 +554,53 @@ void unset()
 	}
 }
 
+int execute(FILE* script, char** argv);
+int _execute(FILE* script, char** argv);
+int collect_command(FILE* script, char** argv);
+
+/* if builtin */
+void if_cmd(FILE* script, char** argv)
+{
+	int index;
+	int status;
+	int old_VERBOSE;
+
+	token = token->next; /* Skip the actual if */
+	/* Do not check for successful exit status */
+	int if_status = _execute(script, argv);
+	old_VERBOSE = VERBOSE;
+	VERBOSE = VERBOSE && !if_status;
+	do {
+		index = collect_command(script, argv);
+		require(index != -1, "Unexpected EOF, improperly terminated if statement.\n");
+		if(0 == index) continue;
+
+		if (0 == if_status) {
+			/* Stuff to exec */
+			status = execute(script, argv);
+		}
+		if (match(token->value, "else")) {
+			if_status = !if_status;
+		}
+	} while (!match(token->value, "fi"));
+	VERBOSE = old_VERBOSE;
```

Comment:
Please use the coding style used throughout kaem/mescc-tools. I think oriansj has an astyle command if you need it